File: ordercoffeeshopfrontend/src/types/product.ts
// src/types/product.ts

export interface ProductVariantRequest {
  sku: string;
  size: string;
  price: number;
  stockQuantity: number;
  isActive: boolean;
}

export interface ProductVariant {
  id: number;
  sku: string;
  price: number;
  stockQuantity: number;
  size: string;
  isActive: boolean;
}

export interface Category {
  id: number;
  name: string;
}

export interface ProductRequest {
  name: string;
  description?: string;
  imageUrl?: string;
  categoryId: number;
  isActive?: boolean;
  variants?: ProductVariantRequest[];
}

export interface Product {
  id: number;
  name: string;
  description: string;
  imageUrl: string;
  isActive: boolean;
  category: Category;
  variants: ProductVariant[];
}

File: ordercoffeeshopbackend/src/main/java/com/coffeeshop/backend/entity/Product.java
package com.coffeeshop.backend.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "products")
@Getter
@Setter
public class Product extends BaseEntity {
    @Column(name = "name", nullable = false, length = 150)
    private String name;

    // Giá có thể được chuyển sang ProductVariant.
    // Giữ lại ở đây nếu bạn muốn có một "giá khởi điểm" hoặc giá mặc định.
    // @Column(name = "price", nullable = false, precision = 10, scale = 2)
    // private BigDecimal price;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "image_url", length = 255)
    private String imageUrl;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id", nullable = true)
    private Category category;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
    private List<ProductVariant> variants = new ArrayList<>();
}
